core/templates/domain/topic/topic-update.service.spec.ts

```diff
@@ -277,77 +277,260 @@ describe('Topic update service', function() {
+      property_name: 'name',
+      new_value: 'new unique value',
+      old_value: 'Topic name'
+    }]);
+  });
 
   it('should set/unset changes to a topic\'s description', function() {
     expect(_sampleTopic.getDescription()).toEqual('Topic description');
-    TopicUpdateService.setTopicDescription(_sampleTopic, 'new description');
-    expect(_sampleTopic.getDescription()).toEqual('new description');
+
+    TopicUpdateService.setTopicDescription(_sampleTopic, 'new unique value');
+    expect(_sampleTopic.getDescription()).toEqual('new unique value');
 
     UndoRedoService.undoChange(_sampleTopic);
     expect(_sampleTopic.getDescription()).toEqual('Topic description');
   });
 
-  it('should create a proper backend change dict for changing descriptions',
-    function() {
-      TopicUpdateService.setTopicDescription(_sampleTopic, 'new description');
-      expect(UndoRedoService.getCommittableChangeList()).toEqual([{
-        cmd: 'update_topic_property',
-        property_name: 'description',
-        new_value: 'new description',
-        old_value: 'Topic description'
-      }]);
-    }
-  );
+  it('should create a proper backend change dict ' +
+  'for changing a topic\'s description',
+  function() {
+    TopicUpdateService.setTopicDescription(_sampleTopic, 'new unique value');
+    expect(UndoRedoService.getCommittableChangeList()).toEqual([{
+      cmd: 'update_topic_property',
+      property_name: 'description',
+      new_value: 'new unique value',
+      old_value: 'Topic description'
+    }]);
+  });
+
+  it('should set/unset changes to a topic\'s abbreviated name', function() {
+    expect(_sampleTopic.getAbbreviatedName()).toEqual(undefined);
+
+    TopicUpdateService.setAbbreviatedTopicName(
+      _sampleTopic, 'new unique value');
+    expect(_sampleTopic.getAbbreviatedName()).toEqual('new unique value');
+
+    UndoRedoService.undoChange(_sampleTopic);
+    expect(_sampleTopic.getAbbreviatedName()).toEqual(undefined);
+  });
+
+  it('should create a proper backend change dict ' +
+  'for changing a topic\'s abbreviated name',
+  function() {
+    TopicUpdateService.setAbbreviatedTopicName(
+      _sampleTopic, 'new unique value');
+    expect(UndoRedoService.getCommittableChangeList()).toEqual([{
+      cmd: 'update_topic_property',
+      property_name: 'abbreviated_name',
+      new_value: 'new unique value',
+      old_value: null
+    }]);
+  });
+
+
+  it('should set/unset changes to a topic\'s thumbnail filename', function() {
+    expect(_sampleTopic.getThumbnailFilename()).toEqual(undefined);
+
+    TopicUpdateService.setTopicThumbnailFilename(
+      _sampleTopic, 'new unique value');
+    expect(_sampleTopic.getThumbnailFilename()).toEqual('new unique value');
+
+    UndoRedoService.undoChange(_sampleTopic);
+    expect(_sampleTopic.getThumbnailFilename()).toEqual(undefined);
+  });
+
+  it('should create a proper backend change dict ' +
+  'for changing a topic\'s thumbnail filename',
+  function() {
```

Comment:
Move this line to the end of the previous one. Ditto elsewhere.

Reply:
Done.